cluster-autoscaler/estimator/binpacking_limit.go

```diff
@@ -0,0 +1,5 @@
+package estimator
+
+type BinpackingLimit interface {
```

Comment:
I think this interface would be better if it exposed both the Node limit by `GetNodeLimit() int` and the Duration limit by `GetDurationLimit() time.Duration`. That way we would be able to implement some duration based limiting easily in the future.

Reply:
I wonder what could be the logic of applying multiple time limits. Could it be same as for the node limits, i.e. take the smallest limit from all defined (>0)?

Currently duration-based limit is set once in the threshold-based limiter constructor method:

```
func NewThresholdBasedEstimationLimiter(
  limits []BinpackingLimit,
  maxDuration time.Duration) EstimationLimiter {...}
}
```

Reply:
Yes, I think we should take the shortest duration limit returned by the limiters. So with pseudo code

```
nodeLimit = min(limit.NodeLimit() for limit in estimator.limits) // excluding zeros
durationLimit = min(limit.DurationLimit() for limit in estimator.limits) // excluding zeros
```

This gives us an advantage of being able to define many duration limits if needed.

Reply:
Thank you - its now implemented in 2742946b9ca375b584f7a2ef2eedd69b8a736fea